src/backend/api/meals.js

```diff
@@ -2,47 +2,116 @@ const express = require('express');
+    res.json({
+      meals: result[0],
+    });
   } catch (error) {
-    throw error;
+    res.status(500).json({ Msg: 'Internal server error' });
   }
 });
+
 router.get('/future-meals', async (req, res) => {
-  const result = await knex.raw(
-    `SELECT * FROM meal ORDER BY created_date DESC `
-  );
-  return res.json({
-    meals: result[0],
-  });
+  try {
+    const result = await knex.raw(
+      `SELECT * FROM meal ORDER BY created_date DESC `
+    );
+    res.json({
+      meals: result[0],
+    });
+  } catch {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
 });
+
 router.get('/past-meals', async (req, res) => {
-  const result = await knex.raw(`SELECT * FROM meal ORDER BY created_date ASC`);
-  return res.json({
-    meals: result[0],
-  });
+  try {
+    const result = await knex.raw(
+      `SELECT * FROM meal ORDER BY created_date ASC`
+    );
+    res.json({
+      meals: result[0],
+    });
+  } catch {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
 });
 
 router.get('/all-meals', async (req, res) => {
-  const result = await knex.raw(`SELECT id, title FROM meal`);
-  return res.json({
-    meals: result[0],
-  });
+  try {
+    const result = await knex.raw(`SELECT id, title FROM meal`);
+    res.json({
+      meals: result[0],
+    });
+  } catch {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
 });
 router.get('/first-meals', async (req, res) => {
-  const result = await knex.raw(`SELECT * FROM meal ORDER BY id LIMIT 1`);
-  return res.json({
-    meals: result[0],
-  });
+  try {
+    const result = await knex.raw(`SELECT * FROM meal ORDER BY id LIMIT 1`);
+    return res.json({
+      meals: result[0],
+    });
+  } catch {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
 });
 router.get('/last-meals', async (req, res) => {
-  const result = await knex.raw(`SELECT * FROM meal ORDER BY id DESC LIMIT 1`);
-  return res.json({
-    meals: result[0],
-  });
+  try {
+    const result = await knex.raw(
+      `SELECT * FROM meal ORDER BY id DESC LIMIT 1`
+    );
+    return res.json({
+      meals: result[0],
+    });
+  } catch {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const result = await knex.insert(req.body).into('meal');
+    res.send(`ID number ${result} is created`);
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+  try {
+    const paramsValue = req.params.id;
+    const result = await knex('meal').where({ id: paramsValue });
+    res.send(result);
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+//Update without hardcore
+router.put('/:id', async (req, res) => {
+  try {
+    const paramsValue = req.params.id;
+    const result = await knex('meal')
+      .where({ id: paramsValue })
+      .update({ price: 80 });
```

Comment:
This would always update the price of the meal to `80`. Similar to the `POST` route you can get the data from the request body and pass it in here.